### PR TITLE
New unit (task 2)

### PR DESCRIPTION
**Pull request by RYQ-22**

### CODE REVIEW

**Review by LazyJazz — CHANGES_REQUESTED** [review]
Fix the nits

**Comment by RYQ-22** [comment]
> Fix the nits
Has been modified.

File: src/battle_game/core/CMakeLists.txt [hunk 1]
```diff
@@ -3,7 +3,7 @@ list(APPEND LIBRARY_LIST ${lib_name})
 
 add_library(${lib_name})
 file(GLOB_RECURSE source_files *.cpp *.h)
-target_sources(${lib_name} PRIVATE ${source_files})
+target_sources(${lib_name} PRIVATE ${source_files} "units/ryq_tank.cpp" "units/ryq_tank.h" "units/ryq_tank2.h" "units/ryq_tank2.cpp" "obstacles/brambles.h" "obstacles/brambles.cpp")
```
Comment (LazyJazz) [inline src/battle_game/core/CMakeLists.txt:6]: remove these

**Review by LazyJazz — CHANGES_REQUESTED** [review]

File: src/battle_game/core/CMakeLists.txt (hunk 1 above)
Comment (RYQ-22) [reply src/battle_game/core/CMakeLists.txt]: Issues have been fixed.

File: src/battle_game/core/units/ryq_tank.h [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  void TankMove(float move_speed, float rotate_angular_speed);
+  void TurretRotate();
+  void Fire();
+  [[nodiscard]] const char *UnitName() const override;
+  [[nodiscard]] const char *Author() const override;
+
+  float turret_rotation_{0.0f};
+  uint32_t fire1_count_down_1{0};
+  uint32_t fire1_count_down_2{0};
+  uint32_t fire1_count_num_{120};
```
Comment (qinchuanhui) [inline src/battle_game/core/units/ryq_tank.h:22]: Had better use `fire1_count_down_1_`. 
Well, just OK this time

**Review by qinchuanhui — APPROVED** [review]
LGTM, you can get the score.
But don't merge it directly due to the format error.

File: src/battle_game/core/CMakeLists.txt [hunk 3]
```diff
@@ -3,7 +3,7 @@ list(APPEND LIBRARY_LIST ${lib_name})
 
 add_library(${lib_name})
 file(GLOB_RECURSE source_files *.cpp *.h)
-target_sources(${lib_name} PRIVATE ${source_files})
+target_sources(${lib_name} PRIVATE ${source_files} "units/ryq_tank.cpp" "units/ryq_tank.h" "units/ryq_tank2.h" "units/ryq_tank2.cpp")
```
Comment (qinchuanhui) [inline src/battle_game/core/CMakeLists.txt:6]: Seems you didn't fix the Cmake issue

**Review by qinchuanhui — CHANGES_REQUESTED** [review]

File: src/battle_game/core/units/ryq_tank.h (hunk 2 above)
Comment (qinchuanhui) [reply src/battle_game/core/units/ryq_tank.h:22]: Sry, you should first fix Cmake error and get the score

File: src/battle_game/core/CMakeLists.txt (hunk 3 above)
Comment (RYQ-22) [reply src/battle_game/core/CMakeLists.txt]: Issues have been fixed.

**Review by qinchuanhui — APPROVED** [review]

**Pull request closed without merging**